### PR TITLE
Resolves #1 unit tests add

### LINKED ISSUE #1: unit tests
to ensure that solutions are correct, and also to help us to develop next exercises

I've looked for both HUnit and HSpec, and imo HSpec is cooler :sunglasses: 

I'll make a PR for this issue

### PR DESCRIPTION
**Pull request by leonardiwagner**
- unit testing add with Hspec
- put exercises into modules (for testing)
- build on travis [build](https://travis-ci.org/leonardiwagner/99questions)

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module Main where
```
Comment (fholiveira) [inline src/Main.hs]: we don't need a main module...

File: test/ListsSpec.hs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    it "Problem 1: Find the last element of a list" $ do
+      mylast [1,2,3,4] `shouldBe` 4
+    it "Problem 3: Find the K'th element of a list. The first element in the list is number 1" $ do
+      kth [1,2,3] 2 `shouldBe` 2
+    it "Problem 4: Find the number of elements of a list" $ do
+      myLength [123, 456, 789] `shouldBe` 3
+    it "Problem 6: Find out whether a list is a palindrome. A palindrome can be read forward or backward; e.g. (x a m a x)" $ do
+      isPalindrome [1,2,3] `shouldBe` False
+      isPalindrome "madamimadam" `shouldBe` True
+    --it "Problem 7: Flatten a nested list structure." $ do
```
Comment (fholiveira) [inline test/ListsSpec.hs]: why this test is commented?

**Comment by fholiveira** [comment]
after fix the issues mentioned in comments, rebase and :sheep:

File: test/ListsSpec.hs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+spec :: Spec
+spec = do
+  describe "Questions 1 to 10: Lists" $ do
+    it "Problem 1: Find the last element of a list" $ do
+      mylast [1,2,3,4] `shouldBe` 4
+    it "Problem 3: Find the K'th element of a list. The first element in the list is number 1" $ do
+      kth [1,2,3] 2 `shouldBe` 2
+    it "Problem 4: Find the number of elements of a list" $ do
+      myLength [123, 456, 789] `shouldBe` 3
+    it "Problem 6: Find out whether a list is a palindrome. A palindrome can be read forward or backward; e.g. (x a m a x)" $ do
```
Comment (aterribili) [inline test/ListsSpec.hs]: Where is the Problem 5 test?

**Comment by fholiveira** [comment]
"one test file to rule them all" :(

File: test/ListsSpec.hs (hunk 2 above)
Comment (leonardiwagner) [inline test/ListsSpec.hs]: I couldn't run this test (dunno why :confused:) , so I updated the PR with a simpler version

File: src/Main.hs (hunk 1 above)
Comment (leonardiwagner) [inline src/Main.hs]: indeed, I changed cabal file from `executable` to `library` now

**Pull request merged**